pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -373,6 +373,24 @@ static char *PyTimeToJSON(JSOBJ _obj, JSONTypeContext *tc, size_t *outLen) {
   return outValue;
 }
 
+static char *PyDecimalToUTF8Callback(JSOBJ _obj, JSONTypeContext *tc,
+                                     size_t *len) {
+  PyObject *obj = (PyObject *)_obj;
+  PyObject *format_spec = PyUnicode_FromStringAndSize("f", 1);
+  PyObject *str = PyObject_Format(obj, format_spec);
+  Py_DECREF(format_spec);
+
+  if (str == NULL) {
+    *len = 0;
+    ((JSONObjectEncoder *)tc->encoder)->errorMsg = "";
```

Comment:
What's the reason for changing this?

Reply:
I assumed that this indicates to the encoder that an error has occurred. Searching for `errorMsg` in the code base indicates this as-well, but I have no golden source supporting my assumption

Reply:
Ah OK - this is all kind of messy so I think OK for this PR, though if you have interest in a follow up could definitely help to clean the error handling. 

I personally don't see a ton of value in straying far from the CPython standards for error handling

Reply:
Agreed. If possible the calling code should just check for a return value of `NULL` instead of this - but that might be quite a big change